app/controllers/admin/erip_transactions_controller.rb

```diff
@@ -6,6 +6,10 @@ class Admin::EripTransactionsController < AdminController
 
   skip_before_action :verify_authenticity_token, only: :bepaid_notify
 
+  http_basic_authenticate_with name: Setting['bePaid_ID'],
+                               password: Setting['bePaid_secret'],
+                               only: :bepaid_notify
+
```

Comment:
Why the logic to make external request to bePaid in controller and not in service?

Reply:
OMG. They actually send the auth parameters back in a webhook. :D

Reply:
Exactly :)